### PR TITLE
Add test for Show[OneAnd]

### PR DESCRIPTION
**Pull request by mikejcurry**
Adds a test for the Show instance for OneAnd

### CODE REVIEW

**Comment by ceedubs** [comment]
:+1: thanks!

It might be nice to have an exact value check for a particular fixed input too.

File: tests/src/test/scala/cats/tests/OneAndTests.scala [hunk 1]
```diff
@@ -47,6 +47,14 @@ class OneAndTests extends CatsSuite {
   checkAll("NonEmptyList[Int]", ComonadTests[NonEmptyList].comonad[Int, Int, Int])
   checkAll("Comonad[NonEmptyList[A]]", SerializableTests.serializable(Comonad[NonEmptyList]))
 
+  test("Show is not empty and is formatted as expected") {
+    forAll { (nel: NonEmptyList[Int]) =>
+      nel.show.nonEmpty should === (true)
+      nel.show.startsWith("OneAnd") should === (true)
```
Comment (adelbertc) [inline tests/src/test/scala/cats/tests/OneAndTests.scala]: Unsure if we want to test specifically for "OneAnd" but I can see there being value in testing if the `.show` of the head is in the string or something

**Comment by mikejcurry** [comment]
I think both of those make sense here. I'll look at them later today.

**Comment by ceedubs** [comment]
:+1:

**Comment by fthomas** [comment]
:+1:

**Pull request merged**